### DISCUSSION TITLE
Alert modal launched by checking checkbox a 3.2.2 failure?

### DISCUSSION THREAD

**Discussion by lamador** [w3c/wcag#4169] [Q&A]
This answer is probably obvious, but is it a 3.2.2 failure when a user tries to check a checkbox and a modal dialog is launched containing an error message? For example, when ordering a pizza there are 10 checkboxes to select specific toppings, but there is a two topping limit. When the user checks a third topping (anchovies), an alert modal is launched, informing the user that the topping limit has been reached. When alert modal is dismissed, the focus returns to the anchovies checkbox which remains in an unchecked state.

I'm hyper-focused on the sentence "Changing the setting of any user interface component is changing some aspect in the control that will persist when the user is no longer interacting with it".  I'm thinking that since the setting of the checkbox hasn't been changed, this is not a 3.2.2 failure.

**Comment by scottaohara** [comment]
per the description you provided, i wouldn't see that as a failure.

**Reply by lamador** [reply]
Thanks Scott

**Comment by TestPartners** [comment]
That sounds like a non-conformance to me because the automatic change of focus to the dialog constitutes a change of context. This is what the normative text refers to. I don’t think the fact that the checkbox setting doesn’t change is relevant.

Steve Green
Managing Director
Test Partners Ltd

From: Lee Amador ***@***.***>
Sent: 09 December 2024 16:16
To: w3c/wcag ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [w3c/wcag] Alert modal launched by checking checkbox a 3.2.2 failure? (Discussion #4169)


This answer is probably obvious, but is it a 3.2.2 failure when a user tries to check a checkbox and a modal dialog is launched containing an error message? For example, when ordering a pizza there are 10 checkboxes to select specific toppings, but there is a two topping limit. When the user checks a third topping (anchovies), an alert modal is launched, informing the user that the topping limit has been reached. When alert modal is dismissed, the focus returns to the anchovies checkbox which remains in an unchecked state.

I'm hyper-focused on the sentence "Changing the setting of any user interface component is changing some aspect in the control that will persist when the user is no longer interacting with it". I'm thinking that since the setting of the checkbox hasn't been changed, this is not a 3.2.2 failure.

**Comment by patrickhlauke** [comment]
not a failure for me, as the change of focus here is caused by the page informing you of a problem - while related to what the user did or didn't do with a user interface component, it's not a behaviour tied directly/always to that UIC. otherwise, a page would never be able to trigger a dialog or alert, ever.

**Comment by TestPartners** [comment]
Pages can trigger dialogs and alerts when buttons or links are operated. Doing so following an attempt to change the value of an input seems so wrong I can’t believe we’re even discussing it. If nothing else, it sounds like lousy interaction design. If the checkbox isn’t supposed to be checked, there are better ways of handling it.

Steve

**Comment by mraccess77** [comment]
I'd typically consider a focus change when you check a checkbox to be a failure. Activating a button  or link is different and thus no notification would be needed for a focus change in that situation at A and AA.
